From 35a5863acf5ee7e1569e803e8471f9d5a51eacae Mon Sep 17 00:00:00 2001
From: Martin Storsjo <martin@bambuser.com>
Date: Fri, 9 Oct 2009 13:59:55 +0300
Subject: [PATCH 1/3] Set the soname to be $(SLIBNAME) instead of $(SLIBNAME_WITH_MAJOR)

Both due to the ways the android linker works, and due to the apk
builder that only includes files named *.so, we want to have the libs
named and refer to each other in the form lib<name>.so.
---
 subdir.mak |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/subdir.mak b/subdir.mak
index 8a407fe..107d37c 100644
--- a/subdir.mak
+++ b/subdir.mak
@@ -36,16 +36,13 @@ define RULES
 $(SUBDIR)%$(EXESUF): $(SUBDIR)%.o
 	$$(LD) $(FFLDFLAGS) -o $$@ $$^ -l$(FULLNAME) $(FFEXTRALIBS) $$(ELIBS)
 
-$(SUBDIR)$(SLIBNAME): $(SUBDIR)$(SLIBNAME_WITH_MAJOR)
-	$(Q)cd ./$(SUBDIR) && $(LN_S) $(SLIBNAME_WITH_MAJOR) $(SLIBNAME)
-
-$(SUBDIR)$(SLIBNAME_WITH_MAJOR): $(OBJS) $(SUBDIR)lib$(NAME).ver
+$(SUBDIR)$(SLIBNAME): $(OBJS) $(SUBDIR)lib$(NAME).ver
 	$(SLIB_CREATE_DEF_CMD)
 	$$(LD) $(SHFLAGS) $(FFLDFLAGS) -o $$@ $$(filter %.o,$$^) $(FFEXTRALIBS) $(EXTRAOBJS)
 	$(SLIB_EXTRA_CMD)
 
 ifdef SUBDIR
-$(SUBDIR)$(SLIBNAME_WITH_MAJOR): $(DEP_LIBS)
+$(SUBDIR)$(SLIBNAME): $(DEP_LIBS)
 endif
 
 clean::
-- 
1.7.3.1

